syntax = "proto3";

package protos;

import "common.proto";

message QueryRequest {
    string table = 1;
    string query = 2;
}

message QueryResponse {
    repeated Row rows = 1;
}


message CountRequest {
    string table = 1;
    string query = 2;
}

message CountResponse {
    int32 result = 1;
}

message BindRequest {
    SegmentId segment_id = 1;
    // string host = 2;
    // int32 port = 3;
}


message UnBindRequest {
    SegmentId segment_id = 1;
}


service Indexer {
    rpc Count(CountRequest) returns (CountResponse) {}
    rpc Query(QueryRequest) returns (QueryResponse) {}
    rpc Bind(BindRequest) returns (Error) {}
    rpc UnBind(UnBindRequest) returns (Error) {}
  }
  